plugins/catalog-backend/src/ingestion/processors/GithubReaderProcessor.ts

```diff
@@ -15,11 +15,29 @@
  */
 
 import { LocationSpec } from '@backstage/catalog-model';
-import fetch from 'node-fetch';
+import fetch, { RequestInit, HeadersInit } from 'node-fetch';
 import * as result from './results';
 import { LocationProcessor, LocationProcessorEmit } from './types';
 
 export class GithubReaderProcessor implements LocationProcessor {
+  private privateToken: string = process.env.GITHUB_PRIVATE_TOKEN || '';
```

Comment:
Managing this token is really tricky, and I think the only "real" way that GitHub provides to do this atm is via GitHub App installations. @benjdlambert is looking into how we could support that in the best way, but in the meantime I think it makes sense to go forward with this.

Reply:
There's a shiny ✨  RFC here: https://github.com/spotify/backstage/issues/1565